workflows/iuf/hooks/master-host-hook-script.yaml

```diff
@@ -0,0 +1,68 @@
+#
+# MIT License
+#
+# (C) Copyright 2022-2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: master-host-hook-script
+spec:
+  entrypoint: main
+  templates:
+    - inputs:
+        parameters:
+          - name: auth_token
+          - name: global_params
+          - name: script_path
+      name: main
+      steps:
+        - - arguments:
+              parameters:
+                - name: dryRun
+                  value: 'false'
+                - name: scriptContent
+                  value: >
+                    set -e
+
+                    global_params_file=$(echo $RANDOM | md5sum | head -c 20; echo)
+
+                    echo "INFO  Saving global_params into file /tmp/${global_params_file}.json"
+
+                    cat << EOF > /tmp/${global_params_file}.json
+
+                    {{inputs.parameters.global_params}}
+
+                    EOF
+
+                    echo "INFO  Calling {{inputs.parameters.script_path}}"
+
+                    chmod +x {{inputs.parameters.script_path}}
+
+                    auth_token="{{inputs.parameters.auth_token}}"
```

Comment:
Does this leak the auth token into the argo logs and/or into the temporary script created by the `ssh-template`?